mapviz_plugins/src/odometry_plugin.cpp

```diff
@@ -278,25 +278,70 @@ namespace mapviz_plugins
       PrintInfo("OK");
     }
   }
+  
+
+  void OdometryPlugin::Paint(QPainter* painter, double x, double y, double scale)
+  {
+    int interval = ui_.show_timestamps->value();
+    if (interval == 0)
```

Comment:
It would be good to explain in a comment *why* you return when `interval == 0`. I suspect it is because we do not wish to paint all the covariance ovals when the interval is zero, but a) a future reader would appreciate that comment; and b) why is 0 the magic number? (For example, I could imagine a reader thinking that an interval of 0 meant to draw every oval, and an interval of 1 meant to draw every other; interval of 2 to draw every third, etc.; this coder would think -1 is the correct magic number for don't-draw-any-ovals.)